core/src/main/java/io/grpc/internal/ManagedChannelImpl.java

```diff
@@ -628,12 +565,12 @@ public void onConnectionClosedByServer(Status status) {
 
                 @Override
                 public void onInUse(TransportSet ts) {
-                  inUseStateAggregator.updateObjectInUse(ts, true);
+                  isIdle = false;
                 }
 
                 @Override
                 public void onNotInUse(TransportSet ts) {
-                  inUseStateAggregator.updateObjectInUse(ts, false);
+                  isIdle = true;
```

Comment:
If its per channel, I could make it a ConcurrentHashMap of booleans, would that work?  I assumed idleness was for the channel, not transport.